### DISCUSSION TITLE
Why doesn't `Collision` implement `Copy` or `Clone`?

### DISCUSSION THREAD

**Discussion by dgunay** [bevyengine/bevy#8120] [Q&A] [answered]
I recently made a simple 2D game with basic collision detection using `bevy::sprite::collide_aabb::Collision`. I wanted to send `Collision`s as events so that downstream systems could react to the collision direction.

I quickly found that because it doesn't implement `Copy` or `Clone`, my only options were to either do everything I needed to do with the `Collision` before moving it into the `EventWriter`, or do an `unsafe` bitwise copy.

`Collision` seems like a very basic enum. Why doesn't it derive these traits?

**Comment by alice-i-cecile** [comment, marked as answer]
This should impl both: can you make a PR?

**Reply by dgunay** [reply]
Gladly: https://github.com/bevyengine/bevy/pull/8121
